### PR TITLE
Added option to create link buttons with URL, instead of just interactive buttons

### LINKED ISSUE #46: Link Action Buttons
Hi,

Are "Link Action Buttons" supported? I can't see how to set the URL for them.

https://api.slack.com/docs/message-attachments#link_buttons

Thanks for the great library 😄

### PR DESCRIPTION
**Pull request by balazscsaba2006**
Closes #46 

Here is a simple example how to create link buttons: https://api.slack.com/docs/message-attachments#link_buttons

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 /.php_cs.cache
 /composer.lock
 /vendor/
+.idea
```
Comment (soullivaneuh) [inline .gitignore]: This has nothing to do here.

File: composer.json [hunk 2]
```diff
@@ -26,7 +26,8 @@
         "guzzlehttp/psr7": "^1.4",
         "mockery/mockery": "^1.0",
         "php-http/message": "^1.6",
-        "php-http/mock-client": "^1.2"
+        "php-http/mock-client": "^1.2",
+        "phpunit/phpunit": "^7.5"
```
Comment (soullivaneuh) [inline composer.json]: Please do not add PHPUnit as a project dependency.

File: composer.json [hunk 3]
```diff
@@ -43,5 +44,8 @@
         "branch-alias": {
             "dev-master": "2.x-dev"
         }
+    },
+    "scripts": {
+        "test": "vendor/bin/phpunit"
```
Comment (soullivaneuh) [inline composer.json]: Custom command is not necessary here.

File: src/AttachmentAction.php [hunk 4]
```diff
@@ -59,6 +59,13 @@ final class AttachmentAction
      */
     private $value;
 
+    /**
+     * Optional value. It can be used to create link buttons
```
Comment (soullivaneuh) [inline src/AttachmentAction.php]: Dot at end of comment.

File: src/AttachmentAction.php [hunk 5]
```diff
@@ -152,6 +159,26 @@ public function setValue(?string $value): self
         return $this;
     }
 
+    /**
+     * @return string|null
```
Comment (soullivaneuh) [inline src/AttachmentAction.php:163]: You can remove this, strict typing already give the information.

File: src/AttachmentAction.php [hunk 6]
```diff
@@ -152,6 +159,26 @@ public function setValue(?string $value): self
...
+     */
+    public function getUrl(): ?string
+    {
+        return $this->url;
+    }
+
+    /**
+     * @param string $url
```
Comment (soullivaneuh) [inline src/AttachmentAction.php]: You can remove this, strict typing already give the information.

File: src/AttachmentAction.php [hunk 7]
```diff
@@ -152,6 +159,26 @@ public function setValue(?string $value): self
...
+     *
+     * @return AttachmentAction
```
Comment (soullivaneuh) [inline src/AttachmentAction.php:173]: You can remove this, strict typing already give the information.

File: src/AttachmentAction.php [hunk 8]
```diff
@@ -179,13 +206,21 @@ public function setConfirm(?ActionConfirmation $confirm): self
...
             'text' => $this->text,
             'style' => $this->style,
             'type' => $this->type,
             'value' => $this->value,
             'confirm' => $this->confirm ? $this->confirm->toArray() : null,
         ];
+
+        // link buttons do not require "name", "value" and "confirm" attributes
+        if (null !== $this->url) {
+            unset($array['name'], $array['value'], $array['confirm']);
```
Comment (soullivaneuh) [inline src/AttachmentAction.php]: Why not start initialization without those attribute and set them after of the URL is not provided.

File: src/AttachmentAction.php [hunk 9]
```diff
@@ -179,13 +206,21 @@ public function setConfirm(?ActionConfirmation $confirm): self
...
+        $array = [
             'name' => $this->name,
             'text' => $this->text,
             'style' => $this->style,
             'type' => $this->type,
             'value' => $this->value,
             'confirm' => $this->confirm ? $this->confirm->toArray() : null,
         ];
+
+        // link buttons do not require "name", "value" and "confirm" attributes
```
Comment (soullivaneuh) [inline src/AttachmentAction.php]: May we have a `@see` tag with slack documentation about this information?

**Review by soullivaneuh — CHANGES_REQUESTED** [review]
Hi and thanks for your contribution. :+1: 

I added some comments to make it merge-able.

File: src/AttachmentAction.php (hunk 5 above)
Comment (balazscsaba2006) [reply src/AttachmentAction.php:163]: I wanted to maintain the exact same styling as the whole class, but I can remove this if you want.

Comment (soullivaneuh) [reply src/AttachmentAction.php:163]: Yeah I understand, the code was created before PHP 7. So unless the CI bot is crying, you can remove it. :+1:

**Comment by soullivaneuh** [comment]
Also, could you please fill the changelog file?

File: src/AttachmentAction.php (hunk 5 above)
Comment (balazscsaba2006) [reply src/AttachmentAction.php:163]: Then it should be removed from the whole file, otherwise there will be no consistency.

**Comment by balazscsaba2006** [comment]
I squashed all the commits into one.

**Comment by balazscsaba2006** [comment]
Upon merging the PR could you please create a release also? Thanks :)

File: src/AttachmentAction.php (hunk 5 above)
Comment (soullivaneuh) [reply src/AttachmentAction.php:163]: Consistency is not a very big deal for this, but anyway let's keep it that way.

If you want to clean files up on this repo next, feel free! :+1:

**Review by soullivaneuh — APPROVED** [review]

**Comment by soullivaneuh** [comment]
Here it is: https://github.com/nexylan/slack/releases/tag/v2.3.0

Thanks again for your work! :+1:

**Pull request merged**